### ISSUE TITLE
Warn if draft creator/submitter is blocked

### ISSUE THREAD

**Issue by AntiCompositeNumber** [WPAFC/afch-rewrite#106]
If the person who created a draft has been blocked, it would be useful for AFCH to warn reviewers like it does for a lack of {{reflist}} or a deleted page. Users might be blocked for sockpuppetry, spam, paid editing, or other reasons that would necessitate a closer review.

**Comment by codeswithroh** [comment]
@enterprisey Can I work in this issue ?

**Comment by Yash-g17** [comment]
Hey @enterprisey  and @AntiCompositeNumber , I'd love to work on this issue 
Can you please assing me ?

**Comment by vartikaSaxena** [comment]
Hey @enterprisey, if this issue is still open I would like to work on it.

**Comment by abarsha** [comment]
hey @enterprisey can i be assigned in this issue? i would love to work with it

**Comment by enterprisey** [comment]
@codeswithroh @Yash-g17 @vartikaSaxena if you're still interested in working on this issue, please add a comment here saying that. If two or more people are interested, the one who posted the original comment earlier has precedence. Or if you want to work together I'd be happy with that too.
@abarsha if none of them respond in a day you can have it

**Comment by Yash-g17** [comment]
@enterprisey I'm still interested

**Comment by abarsha** [comment]
@Yash-g17  hey can we work together?

**Comment by Yash-g17** [comment]
@abarsha Yeah sure , but I'm not sure if this task requires two people , @enterprisey can you please guide us on this

**Comment by abarsha** [comment]
@Yash-g17 well even if this task needs one person other one can have experience and learn things which all we need. I guess it will beneficial for both of us

**Comment by enterprisey** [comment]
yeah, two people will be fine. there are many ways it can be split up. one way we can split it up is one person does the user interface (the message "this user is blocked", and it should also show the block reason), and the other person does the API call. If that sounds good let me know. there are other ways it can be split up, too. we can coordinate on Zulip if you both would like.

**Comment by enterprisey** [comment]
yeah. for this PR you will add a new function checkIsSubmitterBlocked() inside getSubmissionWarnings(), which is in src/modules.submissions.js.
@paul-111 you can start with figuring out which API call you would need to check if a particular user is blocked. for now you can start with the URL (i.e. https://en.wikipedia.org/w/api.php?some=stuff&goes=here). once you get that, try writing the JS to perform the API call and pull out the data we're interested in. examples are all throughout src/modules/submission.js, but try looking at the one at the top of checkDeletionLog().
@Yash-g17 read through the function getSubmissionWarnings and try to figure out how we would write that new function checkIsSubmitterBlocked, following the example of the other four functions already defined there (checkReferences, checkDeletionLog, checkReviewState, checkLongComments). assume you'll already have the data from the API, in particular, four strings: the reason for the block, when they were blocked (in the format "2014-10-08T03:21:32Z"), when the block expires (either "infinity" or a timestamp like for when they were blocked), and the other user who blocked them.

**Comment by paul-111** [comment]
@enterprisey  hey is there any deadline?

**Comment by enterprisey** [comment]
Nope :)

**Comment by Yash-g17** [comment]
@enterprisey I'm done with my part

**Comment by enterprisey** [comment]
@Yash-g17 cool. Can you post your code in this discussion if it's not too long? paul-111 also finished their part and I've asked them to post it here as well. You all are making great progress.

**Comment by DawnArchi01** [comment]
hey ! I see no assignees to this project . I would love to contribute to the project . Since I'm quite new here , it would be really helpful if you can guide whenever necessary . @enterprisey
